Sources/SwiftFusion/Inference/BearingRangeFactor.swift

```diff
@@ -0,0 +1,193 @@
+// Copyright 2019 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import TensorFlow
+//
+///// Protocol for indicating a struct is range-capable
+//public protocol HasRange: Differentiable {
+//  associatedtype A: Differentiable
+//  associatedtype B: Differentiable where B.TangentVector: EuclideanVectorSpace & VectorConvertible & AdditiveArithmetic
+//  @differentiable
+//  static func range(_ from: A, _ to: B) -> Double
+//}
+//
+//public struct PoseToPointRange2D: HasRange {
+//  @differentiable
+//  public static func range(_ from: Pose2, _ to: Vector2) -> Double {
+//    (to - from.t).norm
+//  }
+//}
+//
+//public struct PointToPointRange2D: HasRange {
+//  @differentiable
+//  public static func range(_ from: Vector2, _ to: Vector2) -> Double {
+//    (to - from).norm
+//  }
+//}
+//
+///// Protocol for indicating a struct is bearing-capable
+//public protocol HasBearing: Differentiable {
+//  associatedtype A: LieGroup where A.TangentVector: VectorConvertible
```

Comment:
Similar comment here: You can remove all the constraints except `Differentiable`.